units/host_monitoring_external_termination.go

```diff
@@ -92,58 +92,80 @@ func (j *hostMonitorExternalStateCheckJob) Run(ctx context.Context) {
 		j.env = evergreen.GetEnvironment()
 	}
 
-	settings := j.env.Settings()
+	_, err = HandleExternallyTerminatedHost(ctx, j.ID(), j.env, j.host)
+	j.AddError(err)
+}
 
-	cloudHost, err := cloud.GetCloudHost(ctx, j.host, settings)
-	if err != nil {
-		j.AddError(errors.Wrapf(err, "error getting cloud host for host %s", j.host.Id))
-		return
+// HandleExternallyTerminatedHost will check if a host from a dynamic provider has been termimated
+// and clean up the host if it has. Returns true if the host has been externally terminated
+func HandleExternallyTerminatedHost(ctx context.Context, id string, env evergreen.Environment, h *host.Host) (bool, error) {
```

Comment:
I don't think this should be public and in the units directory. What's calling it outside of this package?

Reply:
I made it private within the units package.